x-pack/test/apm_api_integration/tests/mobile/mobile_detailed_statistics_by_field.spec.ts

```diff
@@ -0,0 +1,132 @@
+import { isEmpty } from 'lodash';
+import moment from 'moment';
+import { APIReturnType } from '@kbn/apm-plugin/public/services/rest/create_call_apm_api';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { generateMobileData, SERVICE_VERSIONS } from './generate_mobile_data';
+
+type MobileDetailedStatisticsResponse =
+  APIReturnType<'GET /internal/apm/mobile-services/{serviceName}/detailed_statistics'>;
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const apmApiClient = getService('apmApiClient');
+  const registry = getService('registry');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+
+  const start = new Date('2023-01-01T00:00:00.000Z').getTime();
+  const end = new Date('2023-01-01T00:15:00.000Z').getTime() - 1;
+
+  async function getMobileDetailedStatisticsByField({
+    environment = ENVIRONMENT_ALL.value,
+    kuery = '',
+    serviceName,
+    field,
+    offset,
+  }: {
+    environment?: string;
+    kuery?: string;
+    serviceName: string;
+    field: string;
+    offset?: string;
+  }) {
+    return await apmApiClient
+      .readUser({
+        endpoint: 'GET /internal/apm/mobile-services/{serviceName}/detailed_statistics',
+        params: {
+          path: { serviceName },
+          query: {
+            environment,
+            start: moment(end).subtract(7, 'minutes').toISOString(),
+            end: new Date(end).toISOString(),
+            offset,
+            kuery,
+            field,
+            fieldValues: JSON.stringify(SERVICE_VERSIONS),
+          },
+        },
+      })
+      .then(({ body }) => body);
+  }
+
+  registry.when(
+    'Mobile detailed statistics when data is not loaded',
+    { config: 'basic', archives: [] },
+    () => {
+      describe('when no data', () => {
+        it('handles empty state', async () => {
+          const response = await getMobileDetailedStatisticsByField({
+            serviceName: 'foo',
+            field: 'service.version',
+          });
+          expect(response).to.be.eql({ currentPeriod: {}, previousPeriod: {} });
+        });
+      });
+    }
+  );
+
+  registry.when(
+    'Mobile detailed statistics when data is loaded',
+    { config: 'basic', archives: [] },
+    () => {
+      before(async () => {
+        await generateMobileData({
+          synthtraceEsClient,
+          start,
+          end,
+        });
+      });
+
+      after(() => synthtraceEsClient.clean());
+
+      describe('when comparison is disable', () => {
+        it('returns current period data only', async () => {
+          const response = await getMobileDetailedStatisticsByField({
+            serviceName: 'synth-android',
+            environment: 'production',
+            field: 'service.version',
+          });
+          expect(isEmpty(response.currentPeriod)).to.be.equal(false);
+          expect(isEmpty(response.previousPeriod)).to.be.equal(true);
```

Comment:
This really needs value-specific tests, not just whether it's empty or not. This is important to make sure we calculate the statistics correctly.

Reply:
100% agree! In general we are a bit light on the value-specific tests (not just this PR).
Afair back when we used ES archives heavily, the recommendation was to NOT use value-specific tests, because it was time consuming to update them when the archive changed.
That is no longer the case, and perhaps we need to broadcast that. 

A tech session on "API test best practices" would be great.